src/relay/transforms/merge_composite.cc

```diff
@@ -122,8 +122,10 @@ class MergeCompositeWrapper : public ExprMutator {
     for (const auto& arg : pattern->args) {
       Expr new_arg;
       if (arg->IsInstance<CallNode>()) {
-        new_arg =
-            ExtractPattern(Downcast<Call>(arg), Downcast<Call>(root->args[i]), var_map, call_map);
+        if (root->args[i]->IsInstance<CallNode>()) {
```

Comment:
Move this condition to the `if` in L124. 

```if (arg->IsInstance<CallNode>() && root->args[i]->IsInstance<CallNode>()) { ... }```

Reply:
if move to L124, then L133  "call_map->Set(arg, new_arg)" will not be executed. I'm not sure if we can bypass the L133 if  arg is CallNode and root->args[i]  is not CallNode.

Reply:
if `root->args[i]->IsInstance<CallNode>()` is false, `new_arg` will be a null pointer. There is no point in caching null pointer.

Reply:
Ideally you should add a test case to verify my claim, but since this change is trivial I'm ok with merging without a test.

Reply:
OK, I have modified.